enclave/core/sgx/snmalloc/snmalloc_wrapper.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/sgxtypes.h>
+#include "../../oe_alloc_thread.h"
+
+namespace snmalloc
+{
+class ThreadAllocUntyped
+{
+  public:
+    static void* get();
+};
+
+void* ThreadAllocUntyped::get()
+{
+    return oe_get_td()->allocator_local;
+}
+} // namespace snmalloc
+
+#define USE_RESERVE_MULTIPLE 1
+#define IS_ADDRESS_SPACE_CONSTRAINED
+
+// In standard configuration, snmalloc defines the oe_* symbols
+// directly. In debug configuration, snmalloc defines oe_nodebug_* symbols
+// instead, and oe_* symbols are directed to the debug shim.
+#if defined(OE_USE_DEBUG_MALLOC)
+#define SNMALLOC_NAME_MANGLE(a) oe_nodebug_##a
+
+extern "C" void* oe_malloc(size_t size)
+{
+    return oe_debug_malloc(size);
+}
+
+extern "C" void* oe_calloc(size_t nmemb, size_t size)
+{
+    return oe_debug_calloc(nmemb, size);
+}
+
+extern "C" void* oe_realloc(void* ptr, size_t size)
+{
+    return oe_debug_realloc(ptr, size);
+}
+
+extern "C" void* oe_memalign(size_t alignment, size_t size)
+{
+    return oe_debug_memalign(alignment, size);
+}
+
+extern "C" void* oe_free(void* ptr, size_t size)
+{
+    return oe_debug_free(ptr, size);
+}
+
+extern "C" size_t oe_malloc_usable_size(void* ptr)
+{
+    return oe_debug_malloc_usable_size(ptr);
+}
+#else
+#define SNMALLOC_NAME_MANGLE(a) oe_##a
+#endif
+// The OpenEnclave runtime manages the thread local allocator pointer, see td_t
+#define SNMALLOC_EXTERNAL_THREAD_ALLOC
+// Enable the OpenEnclave PAL(Platform Abstraction Layer) for OpenEnclave,
```

Comment:
nit: "Open Enclave" should always have a space in the middle (see http://openenclave.io/sdk/).  2 occurrences needing a space in this line.

Reply:
Fixed